src/directives/schema-validate.js

```diff
@@ -1,110 +1,190 @@
-        // Check if our version of angular has i, i.e. 1.3+
-        if (form[attr] && ngModel[attr]) {
-          angular.forEach(form[attr], function(fn, name) {
-            ngModel[attr][name] = fn;
-          });
-        }
-      });
-
-      // Get in last of the parses so the parsed value has the correct type.
-      // We don't use $validators since we like to set different errors depeding tv4 error codes
-      ngModel.$parsers.push(validate);
-
-      // Listen to an event so we can validate the input on request
-      scope.$on('schemaFormValidate', function() {
-        if (ngModel.$setDirty) {
-          // Angular 1.3+
-          ngModel.$setDirty();
-          validate(ngModel.$modelValue);
-        } else {
-          // Angular 1.2
-          ngModel.$setViewValue(ngModel.$viewValue);
+        // Get in last of the parses so the parsed value has the correct type.
+        // We don't use $validators since we like to set different errors depeding tv4 error codes
+        ngModel.$parsers.push(validate);
+
+        // Listen to an event so we can validate the input on request
+        scope.$on('schemaFormValidate', function() {
+          if (ngModel.$setDirty) {
+            // Angular 1.3+
+            ngModel.$setDirty();
+            validate(ngModel.$modelValue);
+          } else {
+            // Angular 1.2
+            ngModel.$setViewValue(ngModel.$viewValue);
+          }
+
+        });
+
+
+        var DEFAULT_DESTROY_STRATEGY = getGlobalOptionsDestroyStrategy();
+
+        function getGlobalOptionsDestroyStrategy() {
+          var defaultStrategy = undefined;
+          if (scope.options && scope.options.hasOwnProperty('destroyStrategy')) {
+            var globalOptionsDestroyStrategy = scope.options.destroyStrategy;
+            var isValidFormDefaultDestroyStrategy = (globalOptionsDestroyStrategy === undefined ||
+                                                    globalOptionsDestroyStrategy === '' ||
+                                                    globalOptionsDestroyStrategy === null ||
+                                                    globalOptionsDestroyStrategy === 'retain');
+            if (isValidFormDefaultDestroyStrategy) {
+              defaultStrategy = globalOptionsDestroyStrategy;
+            }
+            else {
+              console.warn('Unrecognized globalOptions destroyStrategy: %s \'%s\'. Used undefined instead.',
+                  typeof globalOptionsDestroyStrategy, globalOptionsDestroyStrategy);
+            }
+          }
+          return defaultStrategy;
         }
 
-      });
+        // Clean up the model when the corresponding form field is $destroy-ed.
+        // Default behavior can be supplied as a globalOption, and behavior can be overridden in the form definition.
+        scope.$on('$destroy', function() {
+
+          var form = getForm();
+          var conditionResult = $parse(form.condition);
```

Comment:
I was looking through the code and testing some, and I'm not quite sure why we need to check on the condition here? Shouldn't we always just follow the destroy strategy if we get a destroy event? I'm thinking that condition might not be the only thing that can remove a form element from the form, any other add on might use a `ng-if` somewhere and since any field not in the DOM won't get any validation done on it anyways It's probably best to remove it from the model.